util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java

```diff
@@ -204,6 +216,14 @@ public static CredentialFactory getCredentialFactory(
             .setProxyPassword(
                 RedactedString.create(
                     PROXY_PASSWORD_SUFFIX.withPrefixes(keyPrefixes).getPassword(config)))
+            .setAuthClientId(
+                AUTH_CLIENT_ID_SUFFIX.withPrefixes(keyPrefixes).get(config, config::get))
+            .setAuthClientSecret(
+                RedactedString.create(
+                    AUTH_CLIENT_SECRET_SUFFIX.withPrefixes(keyPrefixes).getPassword(config)))
```

Comment:
isn't this already of type `RedactedString`? do we need `RedactedString.create`?